src/ray/core_worker/core_worker_test.cc

```diff
@@ -30,9 +140,77 @@ TEST_F(CoreWorkerTest, TestTaskArg) {
   ASSERT_EQ(*data, *buffer);
 }
 
-TEST_F(CoreWorkerTest, TestAttributeGetters) {
-  ASSERT_EQ(core_worker_.WorkerType(), WorkerType::WORKER);
-  ASSERT_EQ(core_worker_.Language(), Language::PYTHON);
+TEST_F(SingleNodeTest, TestAttributeGetters) {
+  CoreWorker core_worker(WorkerType::DRIVER, Language::PYTHON,
+      raylet_store_socket_names_[0], raylet_socket_names_[0],
+      DriverID::from_random());
+  ASSERT_EQ(core_worker.WorkerType(), WorkerType::DRIVER);
+  ASSERT_EQ(core_worker.Language(), Language::PYTHON);
 }
 
+TEST_F(SingleNodeTest, TestObjectInterface) {
+  CoreWorker core_worker(WorkerType::DRIVER, Language::PYTHON,
+      raylet_store_socket_names_[0], raylet_socket_names_[0],
+      DriverID::from_random());
+  RAY_CHECK_OK(core_worker.Connect());
+
+  uint8_t array1[] = { 1, 2, 3, 4, 5, 6, 7, 8 };
+  uint8_t array2[] = { 10, 11, 12, 13, 14, 15 }; 
+
+  std::vector<LocalMemoryBuffer> buffers;
+  buffers.emplace_back(array1, sizeof(array1));
+  buffers.emplace_back(array2, sizeof(array2));
+
+  std::vector<ObjectID> ids(buffers.size());
+  for (int i = 0; i < ids.size(); i++) {
+    core_worker.Objects().Put(buffers[i], &ids[i]);
+  }
+
+  // Test Get().
+  std::vector<std::shared_ptr<Buffer>> results;
+  core_worker.Objects().Get(ids, 0, &results);
+
+  ASSERT_EQ(results.size(), 2);
+  for (int i = 0; i < ids.size(); i++) {
+    ASSERT_EQ(results[i]->Size(), buffers[i].Size());
+    ASSERT_EQ(memcmp(results[i]->Data(), buffers[i].Data(), buffers[i].Size()), 0);
+  }
+
+  // Test Wait().
+  ObjectID non_existent_id = ObjectID::from_random();
+  std::vector<ObjectID> all_ids(ids);
+  all_ids.push_back(non_existent_id);
+
+  std::vector<bool> wait_results;
+  core_worker.Objects().Wait(all_ids, 2, -1, &wait_results);
+  ASSERT_EQ(wait_results.size(), 3);
+  ASSERT_EQ(wait_results, std::vector<bool>({true, true, false}));
+
+  core_worker.Objects().Wait(all_ids, 3, 100, &wait_results);
+  ASSERT_EQ(wait_results.size(), 3);
+  ASSERT_EQ(wait_results, std::vector<bool>({true, true, false}));
+
+  // Test Delete().
+  // clear the reference held by PlasmaBuffer.
+  results.clear();
+  core_worker.Objects().Delete(ids, true, false);
+
+  // Note that Delete() calls RayletClient::FreeObjects and would not
+  // wait for objects being deleted, so wait a while for plasma store
+  // to process the command.
+  usleep(200 * 1000);
+  core_worker.Objects().Get(ids, 0, &results);
+  ASSERT_EQ(results.size(), 2);
+  ASSERT_TRUE(!results[0] || !results[0]->Data());
+  ASSERT_TRUE(!results[1] || !results[1]->Data());
```

Comment:
Will `Get` return a `nullptr` or `a buffer with nullptr Data` when the object doesn't exist?
It'd be better to make this consistent, i.e., always return nullptr.

Reply:
updated.